libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -92,16 +96,98 @@ export class CartService {
       if (error instanceof CartNotUpdatedError) {
         errorReasonId = CartErrorReasonId.BASIC_ERROR;
       }
-
-      await this.cartManager
-        .finishErrorCart(cartId, {
+      try {
+        await this.cartManager.finishErrorCart(cartId, {
           errorReasonId,
-        })
-        .catch((e) => {
-          // We silently fail here so as not to eat the original error
-          Sentry.captureException(e);
         });
 
+        const cart = await this.cartManager.fetchCartById(cartId);
+        if (cart.stripeSubscriptionId) {
+          const subscription = await this.subscriptionManager.retrieve(
+            cart.stripeSubscriptionId
+          );
+          const invoice = subscription.latest_invoice
+            ? await this.invoiceManager.retrieve(subscription.latest_invoice)
+            : undefined;
+          if (invoice) {
+            switch (invoice.status) {
+              case 'draft':
+                await this.invoiceManager.delete(invoice.id);
+                break;
+              case 'open':
+              case 'uncollectible':
+                await this.invoiceManager.void(invoice.id);
```

Comment:
Is there a chance that there's a time window where Paypal could charge the customer but the invoice still shows draft given the async nature here?

Reply:
Yes, technically. We currently concurrently make the calls to stripe to finalize the invoice and to paypal to charge the customer. If we broke this up and had the calls run in series, we could eliminate the issue. Think its worthwhile?

From the invoice manager:
```
[paypalCharge] = await Promise.all([
  this.paypalClient.chargeCustomer(chargeOptions),
  this.stripeClient.invoicesFinalizeInvoice(invoice.id),
]);
```

Reply:
From call: lets change these calls to occur in series, and add a comment to this effect in the invoice manager